bundle/edu.gemini.itc.web/src/main/java/edu/gemini/itc/web/servlets/ImageServlet.java

```diff
@@ -42,43 +38,28 @@
 
     public static class IdTimedOutException extends RuntimeException {}
 
-    private static final int UpperLimit = 600;
+    private static class LRU extends LinkedHashMap<UUID, ItcSpectroscopyResult> {
+        private static final int CacheLimit = 300;
+        @Override protected boolean removeEldestEntry(final Map.Entry<UUID, ItcSpectroscopyResult> eldest) {
```

Comment:
When and who calls this? Is it called when you add new ones keeping the max size under control?